dvc_http/tests/test_config.py

```diff
@@ -0,0 +1,96 @@
+import ssl
+
+from dvc.fs import HTTPFileSystem
+from fsspec.asyn import get_loop as get_fsspec_loop
+from fsspec.asyn import sync
+
+
+def test_public_auth_method():
+    config = {
+        "url": "http://example.com/",
+        "path": "file.html",
+        "user": "",
+        "password": "",
+    }
+
+    fs = HTTPFileSystem(**config)
+
+    assert "auth" not in fs.fs_args["client_kwargs"]
+    assert "headers" not in fs.fs_args
+
+
+def test_basic_auth_method():
+    user = "username"
+    password = "password"
+    config = {
+        "url": "http://example.com/",
+        "path": "file.html",
+        "auth": "basic",
+        "user": user,
+        "password": password,
+    }
+
+    fs = HTTPFileSystem(**config)
+
+    assert fs.fs_args["client_kwargs"]["auth"].login == user
+    assert fs.fs_args["client_kwargs"]["auth"].password == password
+
+
+def test_custom_auth_method():
+    header = "Custom-Header"
+    password = "password"
+    config = {
+        "url": "http://example.com/",
+        "path": "file.html",
+        "auth": "custom",
+        "custom_auth_header": header,
+        "password": password,
+    }
+
+    fs = HTTPFileSystem(**config)
+
+    headers = fs.fs_args["headers"]
+    assert header in headers
+    assert headers[header] == password
+
+
+def test_ssl_verify_disable():
```

Comment:
Modified `test_ssl_verify_{disable,custom_cert}` to account for changes in https://github.com/iterative/dvc-http/pull/28